pkgs/build-support/rust/default-crate-overrides.nix

```diff
@@ -86,6 +86,13 @@ in
   evdev-sys = attrs: {
     nativeBuildInputs = [ pkg-config ];
     buildInputs = [ libevdev ];
+
+    # upstream has a separate and totally unnecessary codepath for
+    # host!=target which does things like fetching git repositories
```

Comment:
this sounds like an upstream bug, the crate version comes with the submodule but not with `.git`

Reply:
Sorry, the "crate version"?  I don't quite understand.

The "unnecessary codepath" I refer to is this:

https://github.com/ndesh26/evdev-rs/blob/320c6f75ed750bdd3ad9e6c0036f5dc74a8e4c51/evdev-sys/build.rs#L20-L52

Reply:
This line is what I'm referring to:
https://github.com/ndesh26/evdev-rs/blob/320c6f75ed750bdd3ad9e6c0036f5dc74a8e4c51/evdev-sys/build.rs#L54

If you use evdev-sys as a crate, the submodule exists, so the `git submodule update` command doesn't need to be run, but the `.git` directory is absent (probably ignored by `cargo publish`), so it still tries to update anyway. The rest of the function doesn't seem to do any fetching.

Reply:
Oh I see.  You're suggesting that `touch libevdev/.git` might work instead of having to patch the `build.rs`.  I'm trying that right now.  Yes, that would be much less likely to break in the future.

Reply:
It works!

Nice idea, @figsoda.